src/TestSuite/TestCase.php

```diff
@@ -88,6 +94,25 @@ public function skipIf($shouldSkip, $message = '')
     }
 
     /**
+     * update the ErrorLevel.
+     * @param bool $restore restore errorLevel if $restore = true.
+     * @return void
+     */
+    private static function _updateErrorLevel($restore = false)
+    {
+        $needle = 'Cake\Test\TestCase';
+        $class = get_called_class();
+        if (substr($class, 0, 18) !== $needle) {
+            return;
+        }
```

Comment:
Why do you kneed to know the precise class?

Reply:
For not  apply to user code/class
By this if we can ensure this setting just use in core tests
Probably for no break of user tests

Reply:
I means no apply to user test cases